nikolay_shapurov/graph.cpp

```diff
@@ -43,12 +65,62 @@ bool Graph::has_vertex(const VertexId& id) const {
 
 bool Graph::has_edge(const VertexId& from_vertex_id,
                      const VertexId& to_vertex_id) const {
-  for (const auto& from_edge_id : adjacency_map_.at(from_vertex_id)) {
-    for (const auto& to_edge_id : adjacency_map_.at(to_vertex_id)) {
-      if (from_edge_id == to_edge_id) {
-        return true;
+  if (from_vertex_id == to_vertex_id) {
+    for (const auto& from_edge_id : adjacency_map_.at(from_vertex_id)) {
+      for (const auto& to_edge_id : adjacency_map_.at(to_vertex_id)) {
+        const Edge& to_edge = get_edge(to_edge_id);
+        if (to_edge.from_vertex_id == to_edge.to_vertex_id &&
+            from_edge_id == to_edge.from_vertex_id) {
+          return true;
+        }
+      }
+    }
+  } else {
+    for (const auto& from_edge_id : adjacency_map_.at(from_vertex_id)) {
+      for (const auto& to_edge_id : adjacency_map_.at(to_vertex_id)) {
+        if (from_edge_id == to_edge_id) {
+          return true;
+        }
       }
     }
   }
+
+  return false;
+}
+
+bool Graph::has_edge(const EdgeId& id) const {
+  for (const Edge& edge : edges_) {
+    if (id == edge.id) {
+      return true;
+    }
+  }
   return false;
 }
+
+const Vertex& Graph::get_vertex(const VertexId& id) const {
+  assert(has_vertex(id) && "Vertex doesn't exist");
+  const auto it = std::find_if(vertices_.begin(), vertices_.end(),
+                               [id](const Vertex& v) { return v.id == id; });
+  return *it;
+}
+
+const Edge& Graph::get_edge(const EdgeId& id) const {
+  assert(has_edge(id) && "Edge doesn't exist");
+  const auto it = std::find_if(edges_.begin(), edges_.end(),
+                               [id](const Edge& e) { return e.id == id; });
+  return *it;
+}
+
+int Graph::get_vertex_depth(const VertexId& vertex_id) const {
+  const Vertex& v = get_vertex(vertex_id);
+  return v.depth;
```

Comment:
1) Что мы говорили про однобуквенные названия?
2) Тебе вообще переменная тут не нужна
```suggestion
  return get_vertex(vertex_id).depth;
```

Reply:
Исправил